distributed/scheduler.py

```diff
@@ -5983,22 +5986,30 @@ async def restart(self, client=None, timeout=30):
     async def broadcast(
         self,
         comm=None,
-        msg=None,
-        workers=None,
-        hosts=None,
-        nanny=False,
+        *,
+        msg: dict,
+        workers: "list[str] | None" = None,
+        hosts: "list[str] | None" = None,
+        nanny: bool = False,
         serializers=None,
-    ):
+        on_error: "Literal['raise', 'return', 'return_pickle', 'ignore']" = "raise",
+    ) -> dict:  # dict[str, Any]
         """Broadcast message to workers, return all results"""
         parent: SchedulerState = cast(SchedulerState, self)
-        if workers is None or workers is True:
+        if workers is True:
+            warnings.warn(
+                "workers=True is deprecated; pass workers=None or omit instead",
+                category=FutureWarning,
+            )
```

Comment:
Could you add a small `pytest.warns` test for this?

Reply:
Done